controllers/gitrepository_controller_test.go

```diff
@@ -637,6 +637,60 @@ func TestGitRepositoryReconciler_reconcileArtifact(t *testing.T) {
 				*conditions.TrueCondition(meta.ReadyCondition, meta.SucceededReason, "Stored artifact for revision 'main/revision'"),
 			},
 		},
+		{
+			name: "Remove unavailable condition after creating artifact",
+			dir:  "testdata/git/repository",
+			beforeFunc: func(obj *sourcev1.GitRepository) {
+				obj.Spec.Interval = metav1.Duration{Duration: interval}
+				conditions.MarkTrue(obj, sourcev1.ArtifactUnavailableCondition, "Foo", "")
+			},
+			afterFunc: func(t *WithT, obj *sourcev1.GitRepository, artifact sourcev1.Artifact) {
+				t.Expect(obj.GetArtifact()).ToNot(BeNil())
+				t.Expect(obj.GetArtifact().Checksum).To(Equal("b1fab897a1a0fb8094ce3ae0e9743a4b72bd7268"))
+				t.Expect(obj.Status.URL).ToNot(BeEmpty())
+			},
+			want: ctrl.Result{RequeueAfter: interval},
+			assertConditions: []metav1.Condition{
+				*conditions.TrueCondition(meta.ReadyCondition, meta.SucceededReason, "Stored artifact for revision 'main/revision'"),
+			},
+		},
+		{
+			name: "Remove outdated condition after creating new artifact",
```

Comment:
```suggestion
			name: "Removes ArtifactOutdatedCondition after creating new artifact",
```

to be able to distinguish an "outdated condition" from the type :-)